D1371R4.md

```diff
@@ -1362,6 +1363,46 @@ Both use cases are interesting and valid for `inspect`, but resulting `break` be
 Since `inspect` is a new construct and it doesn't have fall-through behaviour that `switch` had, we decided
 that it is more valuable to let `break` to denote iteration stop for enclosing loop.
 
+## Equality vs Identity when matching constant expressions
+
+When evaluating [Expression Pattern] we default to using `operator==` for comparison, rather than checking for bitwise identity.
+
+Imagine following code:
+
+```
+double foo() { return -0.0; }
+
+inspect (foo()) {
+  0.0 => /* ... */;
+  __ => throw Up{};
+}
+```
+
+It will be very surprising for user if the code above won't match `-0.0` to `0.0` and result in exception thrown.
+The intuitive solution is to follow the `operator==` logic and check that the values of the operands are equal.
+
+As a consequence of this decision, users won't be able to simply handle entities that have same value but different representation:
+
+```
+inspect (foo()) {
+  0.0 => /* ... */;
+  -0.0 => /* ... */; // Will never be matched, as 0.0 matches same value.
+  __ => throw Up{};
+}
+```
+
+And would have to use some means to disambiguate those values, by either using `bit_cast` or specialized matcher.
+
+Also note, that similar discussion was held for Non-Type Template Parameters. There the solution was to not use `operator==`, and rely on special set of rules. It was caused by multiple problems caused by "value equality":
```

Comment:
```suggestion
Also note that a similar discussion was held for Non-Type Template Parameters, but there the solution was to rely on special set of rules instead of using `operator==`. This decision was motivated by multiple issues caused by "value equality":
```